pkg/segment/segment.go

```diff
@@ -216,3 +218,37 @@ func IsTelemetryEnabled(cfg *preference.PreferenceInfo) bool {
 	}
 	return false
 }
+
+// sanitizeUserInfo sanitizes username from the error string
+func sanitizeUserInfo(errString string) string {
+	user1, err1 := user.Current()
+	if err1 != nil {
+		return errors.Wrapf(err1, err1.Error()).Error()
+	}
+	errString = strings.ReplaceAll(errString, user1.Username, Sanitizer)
+	return errString
+}
+
+// sanitizeFilePath sanitizes file paths from error string
+func sanitizeFilePath(errString string) string {
+	for _, str := range strings.Split(errString, " ") {
+		if strings.Count(str, string(os.PathSeparator)) > 1 {
+			errString = strings.ReplaceAll(errString, str, Sanitizer)
+		}
+	}
+	return errString
+}
+
+// sanitizeURL sanitizes URLs from the error string
+func sanitizeURL(errString string) string {
+	urlPattern, _ := regexp.Compile(`(http)\S*`)
```

Comment:
Should we consider a regex that matches a URL that might not have prefix "http" as well?
https://stackoverflow.com/questions/42618872/regex-for-website-or-url-validation/42619368